src/error-code.ts

```diff
@@ -686,6 +686,76 @@ export namespace ErrorCodes {
       type: ErrorType.ERROR,
       title: "Unable to find reference out of application function",
     };
+
+  /**
+   * Appsync Integration invocations must be deterministic.
+   *
+   * As stated in the [AppSync Pipeline Resolvers Documents](https://docs.aws.amazon.com/appsync/latest/devguide/pipeline-resolvers.html):
+   *
+   * > Pipeline resolver execution flow is unidirectional and defined statically on the resolver.
+   *
+   * ```ts
+   * const func = new Function(stack, 'id', async () => {});
+   *
+   * new AppsyncResolver(..., async ($context) => {
+   *    // valid
+   *    const f = await func();
+   *    // valid
+   *    await func();
+   *    if($context.arguments.value) {
+   *       // invalid - non-deterministic
+   *       await func();
+   *    }
+   *    while($context.arguments.value) {
+   *       // invalid
+   *       await func();
+   *    }
+   *    // valid
+   *    return func();
+   * });
+   * ```
+   *
+   * Workaround:
+   *
+   * One workaround would be to invoke a lambda function which handles the non-deterministic parts of the workflow.
```

Comment:
Another would be to evaluate an express step function

Reply:
I was trying to be careful about adding too many details to the workarounds as a sfn and a lambda can do most of the same things (except math, logic, etc).